src/darwin/mod.rs

```diff
@@ -334,18 +270,49 @@ pub fn descriptors_for_family(family: &str) -> Vec<Descriptor> {
     out
 }
 
+// The AppleFontSmoothing user default controls font smoothing on macOS. By default it is unset,
+// and the system behaves as though it is set to 2, which means a medium level of font smoothing.
+// We will interpret anything other than 0 as enabling font smoothing, and 0 as disabling it. Note
+// that setting this default as a false boolean value *does not* disable font smoothing; it must be
+// an integer 0.
+static FONT_SMOOTHING_ENABLED: Lazy<bool> = Lazy::new(|| {
+    autoreleasepool(|| unsafe {
+        let key = NSString::alloc(nil).init_str("AppleFontSmoothing");
+        let value: id = msg_send![id::standardUserDefaults(), objectForKey: key];
+
+        if !msg_send![value, isKindOfClass: class!(NSNumber)] {
+            return true;
+        }
+
+        let num_type: id = msg_send![value, objCType];
+        if num_type == nil {
+            return true;
+        }
+
+        // NSNumber's objCType method returns one of these strings depending on the size:
+        // q = quad (long long), l = long, i = int, s = short.
+        // If objCType returns a different specifier the value might be a bool or an unsigned int.
+        let int_specifiers: [&[u8]; 4] = [b"q", b"l", b"i", b"s"];
+        if !int_specifiers.contains(&CStr::from_ptr(num_type as *const i8).to_bytes()) {
+            return true;
+        }
+
+        let smoothing: id = msg_send![value, integerValue];
```

Comment:
Can you remind me what happens when `msg_send!` is called without previously checking for `integerValue`? Does that cause an immediate crash?

Because if it just returns `nil` or something, we don't need to be checking against an array to validate the type.

Reply:
At this point in the code, `value` is confirmed to be a (subclass of) `NSNumber`, and it responded with a non-nil value to `objCType`. Given the possible objCType values I'm checking for, it should have a well-defined response to `integerValue`. I want to check for the `objCType` specifically because of what I mentioned in the comment, that macOS itself requires this value to be an integer to have any effect. `defaults write -g AppleFontSmoothing -bool NO` does not turn smoothing off for anything else, but in that case `integerValue` would 0, and we would disable font smoothing. For bools, `objCType` returns `"c"`, which isn't in the list, so we leave smoothing enabled regardless of the value.

Reply:
Ah interesting, so it would return `0`, that's certainly interesting.